Composer/packages/lib/indexers/src/lgIndexer.ts

```diff
@@ -58,14 +58,11 @@ function index(files: FileInfo[]): LgFile[] {
       const id = getBaseName(name, '.lg');
       const diagnostics = check(content, id);
       let templates: LgTemplate[] = [];
-      if (isValid(diagnostics)) {
-        try {
-          templates = parse(file.content, '');
-        } catch (err) {
-          diagnostics.push(new Diagnostic(err.message, id, DiagnosticSeverity.Error));
-        }
+      try {
```

Comment:
I know there is some confusing and inconvenience how LG library provide features like validation and parsing, the core issue is parsing is not as tolerant as checker and CRUD functions. 

Here is a thread in LG side, that we will consolidate the interface in LG side, and unify the experience, especially on the assumption of when we throw errors, https://github.com/microsoft/botbuilder-dotnet/issues/3084. 

Hopefully with this work, the code in Composer will be simplified significantly.  Please take a look, and leave your opinions.